modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,12 +48,31 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeRealignPeerIds: Option[NonEmptySet[PeerId]] = none
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
-
       private val logger = Slf4jLogger.getLogger[F]
 
+      def realignToMajority(lastSnapshot: Hashed[GlobalIncrementalSnapshot]): F[Unit] =
```

Comment:
Seems like this function is checking the already processed snapshot so it doesn't prevent following a wrong fork. Also it sets the peers to fetch the next snapshot from, but it's not gonna repair the state because the node is already forked. The next snapshots that will be downloaded here will simply be discarded by the snapshot processor because they won't form a correct chain with the last processed one. My suggestion is that we check what the next snapshot should be - first pick the majority based on the realignPeers and only then download the majority snapshot by hash, or download by ordinal but check if the hash matches. We are calculating the majority anyway so it's better to determine before processing the snapshot and not after the snapshot is already processed and the state is broken (or forked rather). We don't need to let the process fork just to fix it later, let's just not let it fork in the first place.

Reply:
I think this has already been resolved, lets confirm here.

Reply:
@ryle-goehausen correct, that code is no longer present